import rick.*
import familiar.*
import error.*

describe "Test de rick"{
	test "Si Rick intenta irse de aventuras con jerry, no puede y falla con una exception generica" {
	
		const unRick = new Rick (demencia = 20 , acompaniante = jerry)
		
		assert.throwsException({unRick.irseDeAventuras()})
	}	
	
	test "Si Rick intenta irse de aventuras con jerry, no puede y falla con el mensaje Soy muy tonto y no puedo irme de aventuras" {

		const unRick = new Rick (demencia = 20 , acompaniante = jerry)
		
		assert.throwsExceptionWithMessage("Soy muy tonto y no puedo irme de aventuras", {unRick.irseDeAventuras()})
	}	
	
	test "Si Rick intenta irse de aventuras con jerry, no puede y falla con la exception ExceptionIrseDeAventuras" {

		const unRick = new Rick (demencia = 20 , acompaniante = jerry)
		
		assert.throwsExceptionWithType(new ExceptionIrseDeAventuras(), {unRick.irseDeAventuras()})
	}	
}